numpy/lib/recfunctions.py

```diff
@@ -807,6 +807,209 @@ def repack_fields(a, align=False, recurse=True):
                      'formats': formats,
                      'titles': titles}, align=align)
 
+def _get_fields_and_offsets(dt, offset=0):
+    """
+    Returns a flat list of (name, dtype, count, offset) tuples of all the
+    scalar fields in the dtype "dt", including nested fields, in left
+    to right order.
+    """
+    fields = []
+    for name in dt.names:
+        sdt = dt.fields[name]
+        count = 1 if sdt[0].shape == () else np.product(sdt[0].shape)
```

Comment:
Why the `if`?

Reply:
Oh yeah I guess product works on empty tuples.